msteams-platform/apps-in-teams-meetings/build-tabs-for-meeting.md

```diff
@@ -176,6 +176,129 @@ TeamsJS is a rich library used to create Tabs using JavaScript. Use the latest T
+
+### Code example
+
+The following code snippet is an example for `readyToUnload` callback function:
+
+```javascript
+// Using readyToUnload callback function
+const beforeUnloadHandler = (readyToUnload: () => void) => {
+  readyToUnload();
+  return true;
+};
+```
+
+The following code snippet is an example of `teamsCore.registerBeforeUnloadHandler` and `teamsCore.registerOnLoadHandler` APIs:
+
+```javascript
+microsoftTeams.app.initialize().then(() => {
+
+    microsoftTeams.teamsCore.registerBeforeUnloadHandler((readyToUnload) => {
+        const result = beforeUnloadHandler(readyToUnload);
+        // dispose resources and then invoke readyToUnload
+        console.log("got beforeunload from TEAMS");
+        return result;
+    });
+
+    microsoftTeams.teamsCore.registerOnLoadHandler((data) => {
+        console.log("got load from TEAMS", data.contentUrl, data.entityId);
+        // use contentUrl to route to correct page 
+        // invoke notifySuccess when ready  
+        app.notifySuccess();
+    });
+});
+```
+
+### Limitations
+
+The following are the limitations for app caching:
+
+* Single-page apps that use client-side routing for page navigation can benefit from app caching. It's recommended that the same domain is used across all contexts of your app launch.
+
+* Apps need to re-register for events such as `themeChange`, `focusEnter`, and so on, in the load handler. Teams client won't send any notifications to the app when cached. If your app requires notifications even when cached, caching might not be the right solution.
+
+* App caching is supported only in Teams desktop client. In Teams web client, even if the app registers load handlers, the app is removed from the cache after the unload sequence is completed.
+
+* Register the `load` and `beforeUnload` handlers early in your launch sequence. If the Teams client doesn’t identify these registrations before the user leaves the app, the app isn't cached.
+
+* The Teams client invokes the `loadHandler` only after the `unload` sequence of the app is completed. For example, if a user launches tab A of your app and then launches tab B of the same app, tab B won't get the load signal until the tab A invokes the `readyToUnload` callback.
+
+* Apps are cached on a per-window basis.
+
+* App caching isn't supported for the meeting stage or Task module contexts, because these can be opened on top of the tab and the same webview can't be used to render the content in the tab and the Task module.
+
+* App caching happens on a per app (not on a per tab) basis within the same window.
+
+* Register only the `beforeUnload` handler if your app doesn't require app caching but needs time to safely save state (if you want to ensure that going away from your app doesn't cause the app content to be abruptly removed from the Document Object Model (DOM)). If the app isn't registered for the `load` event, it's removed from the DOM after the unload flow completes.
```

Comment:
If the app **hasn't** registered the `load` handler

Reply:
Updated as suggested